lang/c++/api/Types.hh

```diff
@@ -69,34 +69,34 @@ inline bool isPrimitive(Type t) {
  * Primitive types are: string, bytes, int, long, float, double, boolean
  * and null
  */
-inline bool isCompound(Type t) {
+inline bool isCompound(Type t) noexcept {
     return (t>= AVRO_RECORD) && (t < AVRO_NUM_TYPES);
 }
 
 /**
  * Returns true if and only if the given type is a valid avro type.
  */
-inline bool isAvroType(Type t) {
+inline bool isAvroType(Type t) noexcept {
     return (t >= AVRO_STRING) && (t < AVRO_NUM_TYPES);
 }
 
 /**
  * Returns true if and only if the given type is within the valid range
  * of enumeration.
  */
-inline bool isAvroTypeOrPseudoType(Type t) {
+inline bool isAvroTypeOrPseudoType(Type t) noexcept {
     return (t >= AVRO_STRING) && (t <= AVRO_NUM_TYPES);
 }
 
 /**
  * Converts the given type into a string. Useful for generating messages.
  */
-AVRO_DECL const std::string& toString(Type type);
+AVRO_DECL const std::string& toString(Type type) noexcept;
 
 /**
  * Writes a string form of the given type into the given ostream.
  */
-AVRO_DECL std::ostream &operator<< (std::ostream &os, avro::Type type);
+AVRO_DECL std::ostream &operator<< (std::ostream &os, avro::Type type) noexcept;
```

Comment:
can't the underlying ostream throw?  Is this guaranteed to be noexcept?

Reply:
Removed noexcept